### PR TITLE
Port App and FatalError components to TS and functional components

### PR DESCRIPTION
**Pull request by joel-jeremy**
Related to https://github.com/actualbudget/actual/issues/1483

### CODE REVIEW

File: packages/desktop-client/src/components/App.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+      }
+    }
+
+    async function initAll() {
+      await Promise.all([installPolyfills(), init()]);
+      setInitializing(false);
+      window.addEventListener('focus', checkScrollbars);
+    }
+
+    initAll().catch(e => console.error(e));
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/App.tsx:99]: **❓ question:** ‏won't this cause issues?

Before: if initialisation fails (for whatever reason) the exception gets thrown and it bubbles up. Eventually breaking the entire app (and thus fatal error message is shown).

Now: if initialisation fails - the exception is shallowed. Do we show an error?

File: packages/desktop-client/src/components/App.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+  async function init() {
+    const socketName = await global.Actual.getServerSocket();
+
+    try {
+      await initConnection(socketName);
+    } catch (e) {
+      if (e.type === 'app-init-failure') {
+        setInitializing(false);
+        return;
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/App.tsx:59]: **💬 suggestion:** ‏I think we should throw an exception here.

Before: `fatalError` gets set and we show error message

Now: `fatalError` is not set - do we show an error? Don't think we do.

**Review by MatissJanis — CHANGES_REQUESTED** [review]
Small regression for mobile views - the footer is cut off:

Screenshot 2023-08-18 at 20 15 05

File: packages/desktop-client/src/components/App.tsx (hunk 2 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/App.tsx]: Updated to show the error UI on init error

File: packages/desktop-client/src/components/App.tsx (hunk 1 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/App.tsx]: Updated to show error UI on init error

**Comment by joel-jeremy** [comment]
The footer in mobile view should now be fixed.

**Review by MatissJanis — APPROVED** [review]
Thanks!

**Pull request merged**